### PR TITLE
Fix/dynamic redux modules

### PR DESCRIPTION
**Pull request by weberjavi**
This PR adds a `sagaRegistry` (similar to `recuderRegistry`) to allow adding sagas to the `sagaMiddleware` once they are needed (instead of all at once on the first load). This avoids the registration of the redux modules imported on those sagas.

I've added the sagas for each module under the `redux-modules` folder.
[PIVOTAL](https://www.pivotaltracker.com/story/show/166605669)

### CODE REVIEW

File: src/store.js [hunk 1]
```diff
@@ -25,7 +26,22 @@ export default () => {
...
   );
-  sagaMiddleware.run(rootSaga)
+  // We set an event listener for the saga registry
+  // So that whenever a new saga gets added
+  // We replace the sagas with the new ones
+  sagaRegistry.setChangeListener((sagas) => {
+    function* allSagas(getState) {
+        yield all(sagas);
+    }
+    sagaMiddleware.run(allSagas);
```
Comment (sorodrigo) [inline src/store.js]: you should probably call cancel on the task that `sagaMiddleware.run` returns before calling run again:
```js
let task;
.
.
.
if (task) {
  task.cancel();
}
task = sagaMiddleware.run(allSagas);
```

**Comment by aabdaab** [comment]
Maybe in this PR we could defer registering `metadata` reducers as well:
image

File: src/store.js (hunk 1 above)
Comment (weberjavi) [reply src/store.js]: Hey @sorodrigo thank you so much for the review  🙇

**Comment by weberjavi** [comment]
> Maybe in this PR we could defer registering `metadata` reducers as well
@aabdaab what do you think about deferring `metadata` and `pageTexts` modules in another PR?
 It is going to require a bigger refactor (info modal is registered on several places where modal opening actions happen, as we are importing actions from the entry file of the modal hence registering it at the same time. We should maybe rethink this pattern to avoid this registration). 
Also some refactor would be needed on the  `About` component because the button that triggers the component is part of the component itself (don't think that's the best approach) so maybe too many things to do here.

**Comment by weberjavi** [comment]
:point_up: ready :)

**Comment by simaob** [comment]
Don't let PRs rot!

**Comment by weberjavi** [comment]
Sorry did not noticed the conflicts after the last `develop` updates :(
Fixed now. Ready again @kfilip94 , @aabdaab ☝️

File: src/pages/data-globe/data-globe-component.jsx [hunk 2]
```diff
@@ -43,7 +48,7 @@ const DataGlobeComponent = ({
   const isBiodiversityActive = activeCategory === 'Biodiversity';
   const isHumanPressuresActive = activeCategory === 'Human pressures';
   const isProtectedAreasActive = activeCategory === 'Existing protection';
-
+console.log(hasMetadata, activeLayers)
```
Comment (kfilip94) [inline src/pages/data-globe/data-globe-component.jsx]: Please remove this console.log

**Comment by kfilip94** [comment]
It looks good :+1:  @weberjavi just remove this console.log and feel free to merge it

**Pull request merged**